sort/challenges-12.test.js

```diff
@@ -0,0 +1,351 @@
+// CHALLENGE 3
+//
+// Write a function named sortByLength that takes in an array of strings and returns the same array,
+// with the strings sorted by their length, lowest to highest.
+// ------------------------------------------------------------------------------------------------
+
+const sortByLength = (strings) => {
+  // Solution code here...
+  return strings.sort((a, b) => a.length > b.length);
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 4
+//
+// Write a function named alphabetizeBetter that takes in an array of strings and returns the same
+// array, with the strings sorted alphabetically. Capitalization should not change the sort order
+// of two strings. For example, ['Alphabet', 'alphabet', 'carrot', 'Zebra'] is correctly sorted,
+// and so is ['alphabet', 'Alphabet', 'carrot', 'Zebra'].
+// ------------------------------------------------------------------------------------------------
+
+const alphabetizeBetter = (strs) => {
+  // Solution code here...
+  return strs.sort((a, b) => {
+    if(a[0] === b[0]){
+      if(a[1].toLowerCase() > b[1].toLowerCase()){
+        return 1;
+      } else {
+        return 0;
+      }
+    } else if(a.toLowerCase() > b.toLowerCase()){
+      return 1;
+    } else {
+      return -1;
+    }
+  });
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 5
+//
+// Write a function named sortByPrice that takes in an array of objects, each of which has a 'price'
+// property, and sorts those objects by price, lowest to highest, returning the same array.
+//
+// Here is an example of the input:
+// [
+//   {name: 'Sweatshirt', price: 45},
+//   {name: 'Bookmark', price: 2.50},
+//   {name: 'Tote bag', price: 15}
+// ];
+// ------------------------------------------------------------------------------------------------
+
+const sortByPrice = (objs) => {
+  // Solution code here...
+  return objs.sort((a, b) => {
+    if(a.price > b.price){
+      return 1;
+    } else if(a.price === b.price){
+      return 0;
+    } else {
+      return -1;
+    } 
+  });
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 6
+//
+// Write a function named sortNumbersByLength that takes in an array of numbers and sorts those numbers
+// by their length.
+//
+// For example, [1, 14, 0.2, -281, 54782] is only correctly sorted in that order.
+// ------------------------------------------------------------------------------------------------
+
+const sortNumbersByLength = (nums) => {
+  // Solution code here...
+  return nums.sort((a, b) => {
+    if(a.toString().length > b.toString().length){
+      return 1;
+    } else if(a.toString().length === b.toString().length){
+      return 0;
+    } else {
+      return -1;
+    }
+  });
+};

```

Comment:
this can be a one-liner also, `return nums.sort( (a,b) => a.toString().length - b.toString().length );`